src/lib.rs

```diff
@@ -598,7 +601,8 @@ impl IntoIter {
         // Make room for another open file descriptor if we've hit the max.
         if self.stack_list.len() - self.oldest_opened == self.opts.max_open {
             self.stack_list[self.oldest_opened].close();
-            self.oldest_opened = self.oldest_opened.checked_add(1).unwrap();
+            // Unwrap is safe here as long as we assume that `self.opts.max_open` is never 0
```

Comment:
If walkdir guarantees that `self.ops.max_open` is never 0, then I can update this comment to indicate that. However, I wrote it this way because, given the arithmetic in the `if` clause, the checked_add will _never_ fail if `self.ops.max_open` is greater than 0. Granted, it's still unrealistic that that bound would ever be reached. So perhaps, yes, I can update the comment to just indicate that.